irispy/new_sji.py

```diff
@@ -0,0 +1,177 @@
+from sunpy import config
+from irispy import iris_tools
+
+__all__ = ['SJICube']
+
+TIME_FORMAT = config.get("general", "time_format")
+
+# the following value is only appropriate for byte scaled images
+BAD_PIXEL_VALUE = -200
+# the following value is only appropriate for unscaled images
+BAD_PIXEL_VALUE_UNSCALED = -32768
+
+
+class SJICube(NDCube):
+    """
+    SJICube
+
+    Class representing SJI Images described by a single WCS
+
+    Parameters
+    ----------
+    data: `numpy.ndarray`
+        The array holding the actual data in this object.
+
+    wcs: `ndcube.wcs.wcs.WCS`
+        The WCS object containing the axes' information
+
+    unit : `astropy.unit.Unit` or `str`
+        Unit for the dataset.
+        Strings that can be converted to a Unit are allowed.
+
+    meta : dict-like object
+        Additional meta information about the dataset.
+
+    uncertainty : any type, optional
+        Uncertainty in the dataset. Should have an attribute uncertainty_type
+        that defines what kind of uncertainty is stored, for example "std"
+        for standard deviation or "var" for variance. A metaclass defining
+        such an interface is NDUncertainty - but isn’t mandatory. If the
+        uncertainty has no such attribute the uncertainty is stored as
+        UnknownUncertainty.
+        Defaults to None.
+
+    mask : any type, optional
+        Mask for the dataset. Masks should follow the numpy convention
+        that valid data points are marked by False and invalid ones with True.
+        Defaults to None.
+
+    extra_coords : iterable of `tuple`s, each with three entries
+        (`str`, `int`, `astropy.units.quantity` or array-like)
+        Gives the name, axis of data, and values of coordinates of a data axis
+        not included in the WCS object.
+
+    copy : `bool`, optional
+        Indicates whether to save the arguments as copy. True copies every
+        attribute before saving it while False tries to save every parameter
+        as reference. Note however that it is not always possible to save the
+        input as reference.
+        Default is False.
+
+    Examples
+    --------
+    >>> from irispy import sji
+    >>> from irispy.data import sample
+    >>> sji = read_iris_sji_level2_fits(sample.SJI_CUBE_1400)
+    """
+
+    def __init__(self, data, wcs, uncertainty=None, unit=None, meta=None,
+                 extra_coords=None, mask=None, copy=False, missing_axis=None):
+        """
+        Initialization of Slit Jaw Imager
+        """
+
+        # Initialize SJI_NDCube.
+        super(SJICube, self).__init__(data, wcs, uncertainty=uncertainty,
+                                      mask=mask, meta=meta, unit=unit,
+                                      extra_coords=extra_coords, copy=copy,
+                                      missing_axis=missing_axis)
+
+    def __repr__(self):
+        return (
+            """
+    SJICube
+    ---------
+    Observatory:\t\t {obs}
```

Comment:
only two tabs here but 3 on most of the other lines? Is that deliberate?

Reply:
Yes, this is to align all datas to have a better global view